tests/test_examples.py

```diff
@@ -809,10 +819,195 @@ def test_multiobject_optimization_prox():
+    eq_lcfs_grid = LinearGrid(rho=1.0, M=4 * eq.M, N=4 * eq.N, NFP=eq.NFP, sym=False)
+
+    field_axis_grid = LinearGrid(
+        rho=1e-2, theta=2 * field.M_well, N=2 * field.N_shift, NFP=field.NFP, sym=False
+    )
+    field_lcfs_grid = LinearGrid(
+        rho=1.0, theta=2 * field.M_well, N=2 * field.N_shift, NFP=field.NFP, sym=False
+    )
+
+    objective = ObjectiveFunction(
+        (
+            Omnigenity(
+                field=field, eq=eq, eq_grid=eq_axis_grid, field_grid=field_axis_grid
+            ),
+            Omnigenity(
+                field=field, eq=eq, eq_grid=eq_lcfs_grid, field_grid=field_lcfs_grid
+            ),
+        )
+    )
+    constraints = (
+        FixModeR(eq=eq),
+        FixModeZ(eq=eq),
+        FixModeLambda(eq=eq),
+        FixPressure(eq=eq),
+        FixIota(eq=eq),
+        FixPsi(eq=eq),
+    )
+
+    optimizer = Optimizer("lsq-exact")
+    (eq, field), _ = optimizer.optimize(
+        (eq, field),
+        objective,
+        constraints,
+        maxiter=100,
+        ftol=1e-6,
+        xtol=1e-6,
+        verbose=3,
+    )
+
+    B_lm = field.B_lm.reshape((field.well_basis.L + 1, -1))
+    B0 = field.well_basis.evaluate(np.array([[0, 0, 0]])) @ B_lm
+    B1 = field.well_basis.evaluate(np.array([[1, 0, 0]])) @ B_lm
+
+    # x_lmn=0 because the equilibrium is QS
+    np.testing.assert_allclose(field.x_lmn, 0, atol=1e-12)
+
+    # check that magnetic well parameters get |B| on axis correct
+    grid = LinearGrid(N=eq.N_grid, NFP=eq.NFP, rho=0)
+    data = eq.compute("|B|", grid=grid)
+    np.testing.assert_allclose(B0, np.mean(data["|B|"]), rtol=1e-3)
+
+    # check that magnetic well parameters get |B| min & max on LCFS correct
+    grid = LinearGrid(M=eq.M_grid, N=eq.N_grid, NFP=eq.NFP, rho=1)
+    data = eq.compute(["min_tz |B|", "max_tz |B|"], grid=grid)
+    np.testing.assert_allclose(np.min(B1), data["min_tz |B|"][0], rtol=2e-3)
+    np.testing.assert_allclose(np.max(B1), data["max_tz |B|"][0], rtol=2e-3)
+
+
+@pytest.mark.regression
+@pytest.mark.solve
+@pytest.mark.slow
+def test_omnigenity_optimization():
+    """Test a realistic OP omnigenity optimization."""
+    # this same example is used in docs/notebooks/tutorials/omnigenity
+
+    # initial equilibrium from QP model
+    surf = FourierRZToroidalSurface.from_qp_model(
+        major_radius=1,
+        aspect_ratio=10,
+        elongation=3,
+        mirror_ratio=0.2,
+        torsion=0.1,
+        NFP=2,
+        sym=True,
+    )
+    eq = Equilibrium(Psi=3e-2, M=4, N=4, surface=surf)
+    constraints = get_fixed_boundary_constraints(eq)
+    eq, _ = eq.solve(objective="vacuum", constraints=constraints, verbose=3)
+
+    # omnigenity optimization
+    field = OmnigenousField(
+        L_well=1,
+        M_well=3,
+        L_shift=1,
+        M_shift=1,
+        N_shift=1,
+        NFP=eq.NFP,
+        helicity=(0, eq.NFP),
+        B_lm=np.array([0.8, 1.0, 1.2, 0, 0, 0]),
+    )
+
+    def mirrorRatio(params):
```

Comment:
are you still planning on making a regular objective for this?

Reply:
It would be nice to have a linear mirror ratio objective for the `B_lm` params, but it's tricky to do in general. Here I initialized the spline knots to be monotonically increasing from B_min to B_max and they happen to stay sorted in that order during the optimization, but that is not guaranteed. 

The easier way to implement this would be through a nonlinear objective on the equilibrium, where we just evaluate |B| on the axis for the Equilibrium (instead of the OmnigenousField)

Reply:
Created Issue #848 to remind us about this. I think it can be done in a future PR.